SampleShared/Samples/TextPanel/DisplayUnityLogs.cs

```diff
@@ -6,23 +6,23 @@
 using System.Text;
 using UnityEngine;
 
-namespace Microsoft.MixedReality.OpenXR.BasicSample
+namespace Microsoft.MixedReality.OpenXR.Sample
 {
     /// <summary>
     /// Prints messages logged through the Unity Console onto a text mesh.
     /// </summary>
-    public class LoggingPanel : MonoBehaviour
+    public class DisplayUnityLogs : MonoBehaviour, ITextProvider
     {
-        [SerializeField]
-        private TextMesh m_text = null;
-
         /// <summary>
         /// The maximum number of lines to be stored for displaying on the text mesh.
         /// If this number is exceeded, old messages will be removed.
         /// </summary>
-        public int MaxLines = 8;
+        public int MaxLines = 20;
 
-        private Queue<string> m_loggedLines = new Queue<string>();
+        private object m_lock = new object();
```

Comment:
Is there a case you're concerned about for needing this lock?

It's nice that this would support logging from background threads, I just don't think we have a use case at the moment. Did you have something in mind, or this is future-proofing?

Reply:
actually, i was wrong.  think about this again, at line 57, we didn't make a copy of the string, so this lock was not actually effective.  so we do rely on the event be triggered on main thread.  it's a bug in my code.  need to fix in next iteration.